tools/autocorres/CorresXF.thy

```diff
@@ -686,7 +686,7 @@ proof -
 qed
 
 (* Merge of lemmas ccorresE and corresXF. *)
-definition "ac_corres st check_termination \<Gamma> rx G \<equiv>
+definition "cancelSignal_corres st check_termination \<Gamma> rx G \<equiv>
```

Comment:
This is definitely a mistake. `ac_corres` is a predicate that is similar in nature to `ccorres_underlying`. It's part of the refinement stack defined by AutoCorres. It's definitely not a `_corres` lemma. :-)

This change and changes to all the lemmas and comments that refer to it will need to be reverted.

Reply:
Will remove the `ac_corres` commit, thanks!

Reply:
I just checked it carefully, is it only the use of `ac_corres` in `autocorres` that's wrong? The refine rename looks fine to me, but the crefine rename looks a bit weird too.

Reply:
Yes, you're right. I had missed that there was a `lemma ac_corres` in each `proof/refine/*/IpcCancel_R.thy` that you've renamed to `lemma cancelSignal_corres`. You'll want to keep that part of the change.

The best way to handle this would be to:
- keep the part of the change that is in `proof/refine`.
- revert the parts of the change in `tools` and `proof/crefine`.

There might be a few more tweaks you need to make, but I think the above will be a good starting point. The main thing is to ensure that the four cases of `lemma ac_corres` in `proof/refine/*/IpcCancel_R.thy` have been renamed, but that the `definition ac_corres` in `tools/autocorres/CorresXF.thy` has *not* been renamed. The typechecker will then force you to get everything else right.